packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -270,8 +269,8 @@ class ThemeData with Diagnosticable {
     BottomNavigationBarThemeData bottomNavigationBarTheme,
     bool fixTextFieldOutlineLabel,
   }) {
-    brightness ??= Brightness.light;
-    final bool isDark = brightness == Brightness.dark;
+    final Brightness _brightness = brightness ?? colorScheme?.brightness ?? Brightness.light;
```

Comment:
This means the user could still specify a `brightness` that doesn't match the given `colorScheme.brightness`. Perhaps if they are both given and don't match we should throw an assertion? Or maybe have the `colorScheme.brightness` take precedence as that is what the eventual value will be?

Reply:
I did the opposite. If you specify both, then the brightness parameter replaces the ColorScheme's brightness. Developers occasionally specify a brightness value in local Theme overrides, to override the the "correct" value, in order to flip light/dark color choices.

Reply:
Ah, then maybe I am missing something because if I set the brightness here that is different from the colorScheme's when do you override the colorScheme's in this constructor? I see where this brightness is used in the constructor to set a bunch of stuff, but then the colorScheme's brightness will be used after that with the new getter.

Reply:
Actually, you're right. It makes more sense to catch this (likely) error than to support it.  I've added the assert and a test for it.